src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -205,6 +225,62 @@ protected function isAttributeToNormalize($object, $attributeName, &$context)
     }
 
     /**
+     * Validates the submitted data and denormalizes it..
+     *
+     * @param string      $currentClass
+     * @param string      $attribute
+     * @param mixed       $data
+     * @param string|null $format
+     * @param array       $context
+     *
+     * @return mixed
+     *
+     * @throws UnexpectedValueException
+     * @throws LogicException
+     */
+    protected function validateAndDenormalize($currentClass, $attribute, $data, $format, array $context)
```

Comment:
it would be better to design an extension point based on composition instead of designing for inheritance. Inheritance-based extension points are always a huge pain to maintain BC when refactoring the logic (see the extraction of name converter from this class for instance)